staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -83,3 +95,30 @@ func (record *attributesRecord) GetOldObject() runtime.Object {
 func (record *attributesRecord) GetUserInfo() user.Info {
 	return record.userInfo
 }
+
+func (record *attributesRecord) SetAnnotations(pluginName string, annotations map[string]string) {
+	if record.annotations == nil {
+		return
+	}
+	for key, value := range annotations {
+		record.SetAnnotation(pluginName, key, value)
+	}
+}
+
+func (record *attributesRecord) SetAnnotation(pluginName, key, value string) {
+	if record.annotations == nil {
+		return
+	}
+	if msgs := validation.NameIsDNSSubdomain(pluginName, false); len(msgs) != 0 {
+		glog.Warningf("plugin_name has invalid format %s", strings.Join(msgs, ","))
+		return
+	}
+	key = pluginName + "/" + key
+
+	record.annotationLock.Lock()
+	defer record.annotationLock.Unlock()
+	if v, ok := record.annotations[key]; ok && v != value {
```

Comment:
I thought we decided we didn't want annotations being overwritten... @tallclair?

Reply:
Annotations in audit should not be overwritten.

Annotations in admission attribute may be overwritten. For example, an admission could make new decision later.